### PR TITLE
fix(database): move document_id secondary indexes to schema sync

### LINKED ISSUE #25697: Strapi v5 migration 5.0.0-06-add-document-id-indexes fails when index already exists or when index name exceeds PostgreSQL identifier limit
### Node Version

v20.19.4

### NPM/Yarn/PNPM Version

10.8.2

### Strapi Version

5.38.0

### Operating System

Other

### Database

PostgreSQL

### Javascript or Typescript

Typescript

### Reproduction URL

_No response_

### Bug Description

The internal migration 5.0.0-06-add-document-id-indexes can fail in PostgreSQL under two conditions:
	1.	When the index already exists but knex.schema.hasIndex() does not detect it.
	2.	When the generated index name exceeds PostgreSQL’s 63 character identifier limit and PostgreSQL truncates the name.

Both cases result in a 42P07 relation already exists error which aborts the transaction and causes Strapi to fail during startup.

### Steps to reproduce

1. Create a Strapi project using PostgreSQL.

2. Before upgrading to a version containing the migration
   `5.0.0-06-add-document-id-indexes`, manually create the index
   expected by the migration:

```sql
CREATE UNIQUE INDEX events_document_id_idx
ON events(document_id);
```

3. Start Strapi so the internal migration runs.

4. During startup, the migration attempts to create the same index again:

```sql
CREATE INDEX "events_document_id_idx" ON "events" ("document_id");
```

5. PostgreSQL returns:

```
ERROR: relation "events_document_id_idx" already exists
errorCode: 42P07
```

### Result

The migration fails and aborts the transaction, causing Strapi to fail during startup.

### Expected result

The migration should detect that the index already exists and skip creation.

---

### Steps to reproduce

1. Use PostgreSQL with Strapi.

2. Ensure a table exists with a long name, for example:

```
strapi_transfer_token_permissions
```

3. The migration generates the index name:

```
strapi_transfer_token_permissions_document_id_locale_published_at_idx
```

4. This name has **69 characters**, exceeding PostgreSQL's identifier limit of **63 characters**.

5. PostgreSQL truncates the identifier and stores the index as:

```
strapi_transfer_token_permissions_document_id_locale_published_
```

6. When the migration runs again, it checks for the original full name and does not detect the truncated index.

7. The migration attempts to create the index again:

```sql
CREATE INDEX "strapi_transfer_token_permissions_document_id_locale_published_at_idx"
ON "strapi_transfer_token_permissions" ("document_id", "locale", "published_at");
```

8. PostgreSQL returns:

```
ERROR: relation "strapi_transfer_token_permissions_document_id_locale_published_" already exists
errorCode: 42P07
```

### Result

The migration fails and aborts the transaction.

### Expected result

Index names should be truncated to PostgreSQL's identifier limit before creation,
or index existence should be detected based on actual stored names.

### Logs

```shell

```

### Code Snippets

_No response_

### Media

_No response_

### Additional information

_No response_

### Confirmation Checklist

- [x] I have checked the existing [issues](https://github.com/strapi/strapi/issues) for duplicates.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/strapi/strapi/blob/develop/CODE_OF_CONDUCT.md).

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Moves document-id secondary index creation out of internal migration `5.0.0-06-add-document-id-indexes` and into model metadata so indexes are created by schema sync.  
Keeps migration `5.0.0-06-add-document-id-indexes` as a stable no-op (same migration name, no DDL).  
Updates tests to cover the new sync-owned index declarations and the migration no-op behavior.

### Why is it needed?

Fixes the fragility reported in #25697 where one-shot migration DDL could fail on duplicate/index-name edge cases and block bootstrap.  
These indexes are schema-level expectations and should be owned by sync, not by a migration that only runs once.

### How to test it?

1. Run targeted tests:
   - `yarn workspace @strapi/database test:unit --testPathPattern='5\\.0\\.0-06-add-document-id-indexes'`
   - `yarn workspace @strapi/core test:unit:vitest -- src/utils/__tests__/transform-content-types-to-models.vitest.test.ts`
2. Verify migration behavior:
   - Confirm `5.0.0-06-add-document-id-indexes` is now a no-op and does not execute index DDL.
3. Verify sync-owned indexes:
   - In `transform-content-types-to-models`, confirm content types declare:
     - single-column `document_id` secondary index
     - one composite index branch based on `locale` / `publishedAt` presence
4. Optional manual bootstrap check:
   - Start Strapi on an upgraded project DB and confirm bootstrap no longer depends on migration 06 DDL for these indexes.

### Related issue(s)/PR(s)

- Fixes #25697
- Fixes CMS-690

### CODE REVIEW

File: packages/core/core/src/utils/__tests__/transform-content-types-to-models.vitest.test.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ const expectedModels = [
     },
     indexes: [
       {
-        name: 'countries_documents_index',
+        name: 'countries_document_id_index',
```
Comment (nclsndr) [inline packages/core/core/src/utils/__tests__/transform-content-types-to-models.vitest.test.ts]: I feel like this could be true 🤔 

### Single-column index rename creates silent duplicate indexes on existing databases

On `develop`, schema sync already writes `{collection}_documents_index` with a **single `document_id` column** for any content type without locale/publishedAt. The PR renames that to `{collection}_document_id_index`.

Schema sync diffs by **name** (see `builder.ts`). Every existing upgraded database will see:

- **Removed:** `countries_documents_index` → `dropIndex` is called
- **Added:** `countries_document_id_index` → `createIndex` is called

`dropIndex` early-returns silently when `forceMigration` is false (the default):

```ts
// packages/core/database/src/schema/builder.ts:205
const dropIndex = (tableBuilder, index, existingIndexes) => {
  if (!db.config.settings?.forceMigration) {
    return; // silent no-op
  }
  ...
};
```

`createIndex` has **no existence guard** — it always fires. So on the default path:
- Old `countries_documents_index` → stays orphaned
- New `countries_document_id_index` → created without issue
- Result: **two single-column indexes on `document_id`**, silently, forever

For users who also hit the old migration 06 DDL (which used `_idx` suffix names like `articles_document_id_idx`), a third overlapping index gets added.

File: packages/core/core/src/utils/transform-content-types-to-models.ts [hunk 2]
```diff
@@ -321,18 +321,43 @@ export const transformContentTypesToModels = (
...
+          name: identifiers.getIndexName([contentType.collectionName, 'documents_locale']),
+          columns: ['documentId', 'locale'].map((n) => identifiers.getColumnName(_.snakeCase(n))),
+        });
+      } else if (hasPublishedAt) {
+        secondaryIndexes.push({
+          name: identifiers.getIndexName([contentType.collectionName, 'documents_published']),
+          columns: ['documentId', 'publishedAt'].map((n) =>
+            identifiers.getColumnName(_.snakeCase(n))
+          ),
+        });
```
Comment (nclsndr) [inline packages/core/core/src/utils/transform-content-types-to-models.ts]: It seems those branches are not covered in tests

File: packages/core/database/src/migrations/internal-migrations/5.0.0-06-add-document-id-indexes.ts [hunk 3]
```diff
@@ -1,80 +1,20 @@
...
-      if (!hasTable) {
-        continue;
-      }
-
-      await addIndexesForTable(knex, meta.tableName);
-    }
+  // Keeping the Umzug signature without doing any DDL; callers pass knex/db for other migrations only.
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars -- required by MigrationFn
+  async up(_knex: Knex.Transaction, _db: Database) {
+    await Promise.resolve();
```
Comment (nclsndr) [inline packages/core/database/src/migrations/internal-migrations/5.0.0-06-add-document-id-indexes.ts]: Do we need to keep a body for the function here?

**Review by nclsndr — COMMENTED** [review]
That's a small but tricky one to test! I did run on v5 but did not v4 > v5.

File: packages/core/core/src/utils/__tests__/transform-content-types-to-models.vitest.test.ts (hunk 1 above)
Comment (innerdvations) [reply packages/core/core/src/utils/__tests__/transform-content-types-to-models.vitest.test.ts]: Good catch, it should be fixed now

**Review by nclsndr — APPROVED** [review]
LGTM 🙌
